### DISCUSSION TITLE
jsonTable date

### DISCUSSION THREAD

**Discussion by Mephisto090** [Zefau/ioBroker.jarvis#1913] [Q&A] [answered]
Hallo,

gibt es eine Möglichkeit ein Datum in einer json via field (Siehe Beispiel) zu formatieren? (Akteulles Format in meiner Json: 2022-09-27T00:00:00+02:00)

```
{"field":"date","label":"Datum","align":"center","sortable":"true"}
```

**Comment by mcuiobroker** [comment, marked as answer]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#parameter-format

`18:10` aktualisiert

**Reply by Mephisto090** [reply]
Kleine Anregung für das Wiki (https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#beispiel-wert-in-spalte-ersetzen)

Kannst du zu diesem Punkt mal ein bisschen was schreiben?

`val === 240 ? 'Geschlossen'`

Was ist das "val", "===" und das Fragezeichen zumBeispiel?

**Reply by mcuiobroker** [reply]
Schau mal bitte in die Doku, reicht das so?
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/besonderheiten-v3/module/jsontable/v3.x.x/parameter-format#beispiel-wert-in-spalte-ersetzen

**Reply by Mephisto090** [reply]
Der Fehlerteufel hat sich eingeschlichen :)

`'format': (val) => val === 241 ? 'Geöffnet' : val`

Ich würde nicht `val` hinter dem `:` schreiben sondern `Geschlossen`. Ich denke das ist leichter zu überblicken als 3x val in dem Code und welches von den dreien meinst du jetzt eigentlich

`'format': (val) => val === 241 ? 'Geöffnet' : 'Geschlossen'`

Nun zum `?`
Ist die Abfrage (val === 241) -> true (wahr) dann wird der Wert val mit dem Wert hinter dem `?` gesetzt, d.h. der Wert val wird dann den Wert 'Geöffnet' erhalten.
Nun zum `:`
Ist die Abfrage (val === 241) -> false (wahr) dann wird der Wert val mit dem Wert hinter dem `?` (ich glaube hier muss ein `:` hin) gesetzt, d.h. der Wert val wird dann den Wert val, also den aktuellen Wert, erhalten.

**Reply by Mephisto090** [reply]
```
val = val==0 ? 0 : 1;
<Bedingung> ? <Ergebnis if true> : <Ergebnis if false>
```

Das `val = val ==` verwirrt ein bisschen :)
Müsste doch eigentlich das erste val weg oder?

**Reply by mcuiobroker** [reply]
Ich hoffe ich habe es jetzt deutlich gemacht, was es bedeutet, schau bitte nochmal.

**Reply by Mephisto090** [reply]
Ich denke jetzt ist es für jeden Verständlich gemacht. Top! Danke
